src/librbd/internal.cc

```diff
@@ -2813,6 +2836,19 @@ int mirror_image_disable_internal(ImageCtx *ictx, bool force,
       return -EINVAL;
     }
 
+    // is mirroring not enabled for the parent? 
+    RWLock::RLocker l(ictx->parent_lock);
```

Comment:
Wrap this section within its own ```{``` ```}``` clause so that the lock is released when you no longer need it.